vivisect/impemu/emulator.py

```diff
@@ -182,10 +182,12 @@ def checkCall(self, starteip, endeip, op):
 
             else:
 
+                apiRetVoid = (ret == None) and (callname != None) \
+                             and (rtype == 'void')
                 if ret == None:
                     ret = self.setVivTaint('apicall', (op,endeip,api,argv))
 
-                callconv.execCallReturn( self, ret, len(funcargs) )
+                callconv.execCallReturn(self, ret, len(funcargs), apiRetVoid)
```

Comment:
again, the callconv object should have all the information necessary, and shouldn't need apiRetVoid handed to it.  if the callconv object *doesn't* know that, that's something else to be fixed.  please look at the problem again, and think about whose responsibility each thing is (anthropomorphically speaking.. sorry, i think of objects as people).

Reply:
Thanks for reviewing.  I will take a look as soon as possible.

Reply:
i now see where i may have contributed to the confusion.  please review my latest comment and let me know if it makes more sense.